src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,159 @@
+package duke;
+
+import duke.command.*;
+
+import java.util.Arrays;
+
+/**
+ * Parser parses and helps to make sense of user input.
+ */
+public class Parser {
+    private final static String INDENTATION = "   ";
+
+    private enum Commands {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND
+    }
+
+    /**
+     * Returns a command to carry out based on the user input.
+     *
+     * @param fullCommand user input.
+     * @return Command to carry out.
+     * @throws DukeException
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        Commands command;
+        String[] userInputs = fullCommand.trim().split(" ");
+        try {
+            command = Commands.valueOf(userInputs[0].toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new InvalidCommandException();
+        }
+        if (userInputs.length == 1) {
+            switch (command) {
+                case BYE:
+                    return new ExitCommand();
+                case LIST:
+                    return new ListCommand();
+                case MARK:
+                    throw new DukeException(INDENTATION +
```

Comment:
I may be wrong but so far I've seen all of your `DukeException`s prefixed with `INDENTATION` (which is just whitespace(?) ). Perhaps you could abstract this padding directly into the constructor of `DukeException`? Again, on the surface, to me it seems like a good piece to abstract out.